### ISSUE TITLE
Javascript C-Dogs?

### ISSUE THREAD

**Issue by cxong** [cxong/cdogs-sdl#208]
This project can compile LLVM bytecode to Javascript:

https://github.com/kripken/emscripten/wiki

Games such as Doom and Dune 2 have been converted to run in the browser. I wonder if it's possible for C-Dogs too?
##

**Comment by meZee** [comment]
I would be interested in seeing C-Dogs [ported](http://kripken.github.io/emscripten-site/docs/porting/emscripten-runtime-environment.html), Emscripten has gained a lot of traction in the last year, and has been featured in the "Mozilla Humble Bundle" in the past. I've heard and read that there are some differences in the way [multithreading](http://flohofwoe.blogspot.com/2013/01/multithreading-in-emscripten-with-html5.html) is handled in Emscripten, which I'm not sure is even an issue for C-Dogs, the other common issues are listed here, in [API Limitations](http://kripken.github.io/emscripten-site/docs/porting/guidelines/api_limitations.html). While we're at it, I wouldn't mind seeing a [PNaCL & NaCL](https://developer.chrome.com/native-client) port as well. It's something to look into.

**Comment by cxong** [comment]
There is indeed no multithreading in C-Dogs. However, there's quite a bit of file I/O (graphics, missions, load/save config files), which might make porting more work.

I did briefly look at porting myself a year ago; I tried with a [very simple console program](https://github.com/cxong/CyberdogsNameGen) (stdout only) so I know it works, but when I tried with C-Dogs I couldn't figure out what was wrong or what I needed to do. Ideally someone with prior experience doing these ports can take a look.
